akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -284,6 +292,55 @@ import akka.stream.stage._
     override def toString: String = "SingleSource"
   }
 
+  final class IterableSource[T](val elements: immutable.Iterable[T]) extends GraphStage[SourceShape[T]] {
+    ReactiveStreamsCompliance.requireNonNullElement(elements)
+    override protected def initialAttributes: Attributes = DefaultAttributes.iterableSource
+    private val out = Outlet[T]("IterableSource.out")
+    override val shape: SourceShape[T] = SourceShape(out)
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+      new GraphStageLogic(shape) with OutHandler {
+        lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
+        private var iterator: Iterator[T] = _
+
+        override def preStart(): Unit = {
+          iterator = elements.iterator
+        }
+
+        override def onPull(): Unit =
+          try {
+            if (iterator.hasNext) {
+              push(out, iterator.next())
+              if (!iterator.hasNext) {
+                completeStage()
+              }
+            } else {
+              completeStage()
+            }
+          } catch handleException
+
+        private def handleException: Catcher[Unit] = {
```

Comment:
since this is only used in one place it can be an ordinary try-catch instead of `Catcher`